python/cudf/cudf/testing/testing.py

```diff
@@ -234,6 +257,12 @@ def assert_column_equal(
         )
 
 
+def null_safe_scalar_equals(left, right):
+    if left in {cudf.NA, np.nan} or right in {cudf.NA, np.nan}:
+        return left is right
+    return left == right
```

Comment:
Could we just do 
```suggestion
    return (left is right) or (left == right)
```
?

Reply:
The problem with this approach is that if `left == NaN` and `right == NA`, the first check evaluates false and the second one evaluates to `NA`, making the whole expression evaluate to `NA`. This ends up creating an `if NA:` statement from the output and triggers a TypeError. This causes the `diff` to read 100% and we get the wrong error message. The original function avoids this situation by never allowing `NA` or `NaN` to pass to the `==` check. I added a test

Reply:
Gotcha, yeah that makes sense.